doc/source/cd/background.ipynb

```diff
@@ -0,0 +1,608 @@
+    "$$\n",
+    "\n",
+    "with the standard deviation set at $\\sigma=0.8$, and the weights set to $\\phi_1=\\phi_2=0.5$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sigma = 0.8\n",
+    "train_norm_0 = sp.stats.multivariate_normal([-1,-1], np.eye(2)*sigma**2)\n",
+    "train_norm_1 = sp.stats.multivariate_normal([ 1, 1], np.eye(2)*sigma**2)\n",
+    "\n",
+    "X_0 = train_norm_0.rvs(size=120,random_state=0)\n",
+    "X_1 = train_norm_1.rvs(size=120,random_state=0)\n",
+    "X_train = np.vstack([X_0, X_1])\n",
+    "\n",
+    "true_slope = -1\n",
+    "y_train = true_model(X_train,true_slope)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "source": [
+    "To save time later on, a plotting function is defined:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "outputs": [],
+   "source": [
+    "plt.rcParams.update({'font.size': 16})\n",
+    "\n",
+    "def plot(X,y, slope, clf=None):\n",
+    "    # Init plot\n",
+    "    fig, ax = plt.subplots(figsize=(4,4))\n",
+    "    #ax.axis('equal')\n",
+    "    ax.set_xlabel(r'$x_1$')\n",
+    "    ax.set_ylabel(r'$x_2$')\n",
+    "    ax.set_xlim([-3.5,3.5])\n",
+    "    ax.set_ylim([-6,3.3])\n",
+    "    \n",
+    "    # Plot data\n",
+    "    cmap = LinearSegmentedColormap.from_list('my_cmap', [(0.78,0.44,0.22), (0.22, 0.44, 0.78)], N=2)\n",
+    "    scat = ax.scatter(X[:,0], X[:,1],c=y,ec='k',s=70,cmap=cmap,alpha=0.7)\n",
+    "    \n",
+    "    # Plot true decision boundary\n",
+    "    xconcept = np.array(ax.get_xlim())\n",
+    "    ax.plot(xconcept,xconcept*slope,'k--',lw=2,alpha=0.8)\n",
+    "    \n",
+    "    if clf is not None:\n",
+    "        # Plot classifier decision boundary\n",
+    "        xlim = ax.get_xlim()\n",
+    "        ylim = ax.get_ylim()\n",
+    "\n",
+    "        xx = np.linspace(xlim[0], xlim[1], 100)\n",
+    "        yy = np.linspace(ylim[0], ylim[1], 100)\n",
+    "        XX, YY = np.meshgrid(xx, yy)\n",
+    "        Z = clf.predict(np.c_[XX.ravel(), YY.ravel()])\n",
+    "\n",
+    "        # Put the result into a color plot\n",
+    "        Z = Z.reshape(XX.shape)\n",
+    "        ax.contourf(xx, yy, Z, cmap=cmap, alpha=.3, levels=1)\n",
+    "        ax.contour(xx, yy, Z, colors='k', linestyles=':', linewidths=[2,0], levels=1,alpha=0.8)\n",
+    "\n",
+    "    plt.show()\n",
+    "    \n",
+    "labels = ['No','Yes']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For our model, we choose the well-known, but simple, decision tree classifier. As well as training the model, this is a good time to initialise and \"train\" the MMD detector by calling:\n",
+    "\n",
+    "```python\n",
+    "detector = MMDDrift(X_train, backend='tensorflow', p_val=.05)\n",
```

Comment:
Might be good to mix it up with a `pytorch` backend sometimes just to highlight that this is also possible.